aesara/tensor/basic.py

```diff
@@ -987,7 +989,9 @@ def zeros(shape, dtype=None):
 
 def ones(shape, dtype=None):
     """Create a `TensorVariable` filled with ones, closer to NumPy's syntax than ``alloc``."""
-    if not isinstance(shape, (np.ndarray, Sequence, TensorVariable)):
+    if (not isinstance(shape, (np.ndarray, Sequence, TensorVariable))) or (
+        isinstance(shape, TensorVariable) and len(shape.shape.eval()) == 0
+    ):
```

Comment:
This is still using the `shape.eval()`

Reply:
Yeah, will update this one and push again :D